contracts/distribution/AlphaReleaseRuleSelector.sol

```diff
@@ -12,22 +13,22 @@ import "../interfaces/IAlphaReleaseRuleSelector.sol";
  **/
 
 contract AlphaReleaseRuleSelector is Ownable, IAlphaReleaseRuleSelector {
-  address constant HEAD = address(1);
+  using SafeMath for uint256;
 
   /**
-   * @dev the receiver linked list
-   *  receiver address => the next receiver address
+   * @dev the struct for storing the receiver's rule
    */
-  mapping(address => address) public receiverList;
-  /**
-   * @dev the mapping of receiver address to the Alpha release rule
-   *  receiver address => the Alpha release rule
-   */
-  mapping(address => IAlphaReleaseRule) public rules;
+  struct ReceiverRule {
+    // Alpha receiver
+    IAlphaReceiver receiver;
+    // release rule
+    IAlphaReleaseRule rule;
+  }
+
   /**
-   * @dev the number of all rules registered to the Alpha release rule selector
+   * @dev the list of receivers with rule
    */
-  uint256 public ruleCount;
+  ReceiverRule[] public receiverRuleList;
```

Comment:
Add view function to get the length of list.

Reply:
Fixed